spot_driver/spot_driver/spot_ros2.py

```diff
@@ -844,6 +844,31 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
                 if self.auto_stand.value:
                     self.spot_wrapper.stand()
 
+        # Explicitly get a lease when get_lease_on_action is False
+        # because spot_wrapper does not get a lease at initialization.
+        if self.spot_wrapper is not None and not self.get_lease_on_action.value:
+            self.spot_wrapper.getLease()
+        self.create_service(
+            srv_type=Trigger,
+            srv_name="gen_new_spot_ros_lease",
+            callback=self.new_lease_callback,
```

Comment:
Maybe:
```suggestion
            srv_name="get_lease",
            callback=self.get_lease_callback,
```

to keep things consistent and make it clear what the service is calling under the hood?

Reply:
I'm in favor of consistency. I'd also like to make it clear that this is not a "getter" in the sense of just returning some information, but also triggers what BD API calls either "take" or "acquire":

https://github.com/bdaiinstitute/spot_wrapper/blob/5bcf6d53abc6c9ae2fe8dd9ef8cadfeb1ca52e60/spot_wrapper/wrapper.py#L1070

question: do we want to expose to ROS these two different "get"s individually?

Reply:
@ name of service
I think `get_lease` could be confusing, especially in the context of the semiautonomy setting where many different nodes are "taking" the lease (generating new leases) -- someone might think `get_lease` doesn't generate a new lease, when it actally does. 

What about `spot_wrapper_get_lease`, or just something that references spot_wrapper somehow to make it clearer that it's spot_wrapper's get_lease?

@ exposing two different "get"s 
Maybe we can wait until someone has a use case for getting the current lease `SpotROS` is using, without generating a new one? I don't see anyone needing this in a semiautonomy setting where only one ros node can command Spot at a time. Will there be a scenario where we want two ros nodes to be able to command Spot simultaneously, where one commands it via `SpotROS`, and another commands it via its own `RobotCommandClient`? In such a case, the latter node might want to use the same lease `SpotROS` is using, and so will want to ask for it via a "get". Perhaps such a scenario is if we want a navigation node to control Spot's legs, while some other node control's Spot's arm? Would we want to do anything like this any time soon? If not, we can not expose both "get"s.

Reply:
Answered the first 2 parts below - I am ok with changing the function in spot_wrapper, so long as we make sure both spot_ros2 and spot_ros are fixed (will only need to check with the maintainer of spot_ros to see if there are any issues on their end).

As for the last paragraph, I don't think we need that until we have a use case. If you have 2 nodes that want to control a spot then they can both go through one `spot_driver`. Planning for an alternative interface from ros to the spot-sdk is outside the scope of this repo.